plugins/modules/ndo_ipsla_monitoring_policy.py

```diff
@@ -0,0 +1,461 @@
+
+__metaclass__ = type
+
+ANSIBLE_METADATA = {
+    "metadata_version": "1.1",
+    "status": ["preview"],
+    "supported_by": "community",
+}
+
+DOCUMENTATION = r"""
+---
+module: ndo_ipsla_monitoring_policy
+short_description: Manage IPSLA Monitoring Policies on Cisco Nexus Dashboard Orchestrator (NDO).
+description:
+- Manage IPSLA Monitoring Policies on Cisco Nexus Dashboard Orchestrator (NDO).
+- This module is only supported on ND v3.1 (NDO v4.3) and later.
+author:
+- Shreyas Srish (@shrsr)
+options:
+  template:
+    description:
+    - The name of the template.
+    - The template must be a tenant template.
+    type: str
+    required: true
+  ipsla_monitoring_policy:
+    description:
+    - The name of the IPSLA Monitoring Policy.
+    type: str
+    aliases: [ name ]
+  description:
+    description:
+    - The description of the IPSLA Monitoring Policy.
+    type: str
+  ipsla_monitoring_policy_uuid:
+    description:
+    - The uuid of the IPSLA Monitoring Policy.
+    - This parameter is required when the O(ipsla_monitoring_policy) needs to be updated.
+    type: str
+    aliases: [ uuid ]
+  sla_type:
+    description:
+    - The type of SLA.
+    type: str
+    choices: [ http, tcp, icmp, l2ping ]
+  destination_port:
+    description:
+    - The port used for SLA.
+    type: int
+  http_version:
+    description:
+    - The HTTP version used for SLA.
+    type: str
+    choices: [ HTTP10, HTTP11 ]
+  http_uri:
+    description:
+    - The URI used for HTTP SLA.
+    type: str
+  sla_frequency:
+    description:
+    - The frequency of SLA monitoring.
+    type: int
+  detect_multiplier:
+    description:
+    - The detection multiplier for SLA.
+    type: int
+  req_data_size:
+    description:
+    - The size of the request data.
+    type: int
+  type_of_service:
+    description:
+    - The IPv4 Type of Service.
+    type: int
+  operation_timeout:
+    description:
+    - The operation_timeout for SLA.
+    type: int
+  threshold:
+    description:
+    - The threshold for SLA.
+    type: int
+  ipv6_traffic_class:
+    description:
+    - The IPv6 Traffic Class.
+    type: int
+  state:
+    description:
+    - Use C(absent) for removing.
+    - Use C(query) for listing an object or multiple objects.
+    - Use C(present) for creating or updating.
+    type: str
+    choices: [ absent, query, present ]
+    default: query
+extends_documentation_fragment: cisco.mso.modules
+"""
+
+EXAMPLES = r"""
+- name: Create a new IPSLA monitoring policy
+  cisco.mso.ndo_ipsla_monitoring_policy:
+    host: mso_host
+    username: admin
+    password: SomeSecretPassword
+    template: ansible_tenant_template
+    ipsla_monitoring_policy: ansible_test_ipsla_monitoring_policy
+    sla_type: http
+    http_version: HTTP11
+    http_uri: /test
+    sla_frequency: 120
+    detect_multiplier: 5
+    req_data_size: 64
+    type_of_service: 16
+    operation_timeout: 1000
+    threshold: 1000
+    ipv6_traffic_class: 32
+    state: present
```

Comment:
Wouldn't it better for this example to make the sla_type to TCP and therefore add the destination_port attribute? Like that, we have a full example with all attributes being used.

Reply:
If we use TCP http_version, http_uri are not required.